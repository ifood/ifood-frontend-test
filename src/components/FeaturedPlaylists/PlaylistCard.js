import React from 'react';

const PlaylistCard = (props) => (
  <div className="col-4">
    <div className="card">
    <img className="card-img-top" src={props.imageUrl} alt={props.name} />
    <div className="card-body">
      <h5 className="card-title">{props.name}</h5>
      <p className="card-text">Total tracks: {props.tracks.total}</p>
      <a href={props.externalUrl} className="btn btn-primary">See on Spotify</a>
    </div>
  </div>
  </div>
);

export default PlaylistCard;