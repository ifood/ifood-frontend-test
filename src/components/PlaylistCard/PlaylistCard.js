import React from 'react';
import Proptypes from 'prop-types';

const PlaylistCard = ({
  href,
  name,
  image,
}) => ((
  <div className="card">
    <div className="card-image">
      <a href={href} target="_blank">
        <figure className="image is-4by3">
          <img src={image} alt="playlist picure" />
        </figure>
        <span className="card-title">{name}</span>
      </a>
    </div>
  </div>
));

PlaylistCard.propTypes = {
  href: Proptypes.string,
  name: Proptypes.string,
  image: Proptypes.string,
};

export default PlaylistCard;
