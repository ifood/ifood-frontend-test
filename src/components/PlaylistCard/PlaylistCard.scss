.playlist-card {
  position : relative;
  margin-bottom: 1em;

  &__image {
    height: 300px;
  }

  &__content {
    position: absolute;
    bottom: 0;
    width: 100%;
    background: rgba(0, 0, 0, .7);

    h2 {
      color: #fff;
      font-size: 1.5em;
    }
  }
}
