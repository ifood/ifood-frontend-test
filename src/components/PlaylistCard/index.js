import React from "react";

export default function PlaylistCard({ playlist }) {
  const { images, collaborative, public } = playlist;
  return (
    <div
      tabindex="0"
      role="button"
      className="playlist-card"
      aria-label={`Playlist named ${playlist.name}`}
    >
      <div className="playlist-card__dark-cover" />
      <img className="playlist-card__img" src={images[0].url}></img>
    </div>
  );
}
