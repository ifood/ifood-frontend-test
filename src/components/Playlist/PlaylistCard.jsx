import React from 'react';

import styled from 'styled-components';
import { randomColors, mainColor, white, black } from '../style/colors';

export const Card = styled.div`
    position: relative;
    width: 200px;
    height: 200px;
    display: flex;
    flex: 0 1 auto;
    align-content: flex-start;
    padding: 20px;
    flex-flow: column nowrap;
    justify-content: space-around;
    background: ${(props) => props.backgroundColor || mainColor};
    margin-bottom: 20px;

    div {
        flex-grow: 1;
        margin-right: 5px;

        &:last-child {
            margin-right: 0;
        }
    }

    @media (max-width: 1024px) {
        width: 100%;

        div {
            margin-right: 0;
        }
    }
`;

export const CardImage = styled.div`
    width: 80px;
    height: 80px;
    background: url(${(props) =>
        props.backgroundImage || 'https://dummyimage.com/80x80/000/fff&text=x'});
    background-repeat: no-repeat;
    background-size: 80px 80px;
    position: absolute;
    top: 20px;
    left: 20px;
    opacity: 0.6;

    @media (max-width: 1024px) {
        opacity: 1;
    }
`;

export const Name = styled.h1`
    font-family: 'Oswald', sans-serif;
    font-size: 36px;
    font-weight: 700;
    text-transform: uppercase;
    text-align: right;
    color: ${white};
`;

export const Author = styled.h1`
    font-family: 'Oswald', sans-serif;
    font-size: 16px;
    text-align: right;
    color: ${black};
`;

export const Description = styled.p`
    margin-top: 5px;
    font-family: 'Noto Sans JP', sans-serif;
    font-size: 16px;
    color: ${white};
`;

const PlayListCard = ({ name, description, owner, images }) => {
    const { display_name: displayName, href } = owner;
    const backgroundImage = images[0].url;
    const randomNumber = Math.floor(Math.random() * 10);

    return (
        <Card backgroundColor={randomColors[randomNumber]}>
            <CardImage backgroundImage={backgroundImage} />
            <Name>{name}</Name> <Author>by {displayName}</Author>
            <Description>{description}</Description>
        </Card>
    );
};

export default PlayListCard;
