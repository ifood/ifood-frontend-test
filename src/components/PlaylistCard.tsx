import { FontAwesomeIcon as Icon } from '@fortawesome/react-fontawesome';
import React, { Fragment, PureComponent } from 'react';
import Img from 'react-image';
import styled from 'styled-components';

import { IPlaylist } from '../api/spotify';

interface IProps {
  playlist: IPlaylist;
}

const CARD_SIZE = '15rem';

const StyledCardContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: 1.6em;
  width: ${CARD_SIZE};

  /* truncate playlist name */
  .playlist-name {
    width: ${CARD_SIZE};
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    font-size: 1.6em;
  }
`;

const StyledImageContainer = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;

  height: ${CARD_SIZE};
  width: ${CARD_SIZE};

  & img {
    max-height: ${CARD_SIZE};
    max-width: ${CARD_SIZE};
  }
`;

const DiscPlaceholder = () => (
  <Icon className='has-text-light' icon='compact-disc' size='10x' />
);

export default class PlaylistCard extends PureComponent<IProps> {
  public render() {
    const { playlist } = this.props;
    return (
      <StyledCardContainer>
        <a href={playlist.external_urls.spotify} target='_blank'>
          <StyledImageContainer>
            <Img
              className='image'
              src={playlist.images[0].url}
              loader={<DiscPlaceholder />}
              unloader={'failed to load image'}
            />
          </StyledImageContainer>
        </a>
        <div
          className='playlist-name has-text-centered has-text-weight-light'
          title={playlist.name}
        >
          {playlist.name}
        </div>
      </StyledCardContainer>
    );
  }
}
