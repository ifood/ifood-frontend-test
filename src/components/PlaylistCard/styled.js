import styled from "styled-components";

import * as C from "constants/styles/colors";
import * as T from "constants/styles/typography";
import { SPACINGS } from "constants/styles/spacings";

export const PlaylistCardLink = styled.span`
  color: ${C.PRIMARY_COLOR};
  margin: ${SPACINGS.xl} 0 auto;
  text-align: center;
  font-weight: bold;
  font-size: ${T.SIZES.sm};
  padding: ${SPACINGS.sm} 0 ${SPACINGS.sm};
  border-bottom: 5px solid currentColor;
  display: block;
  transition: 300ms ease all;
`;

export const PlaylistCard = styled.a`
  width: 100%;
  max-width: 300px;
  display: flex;
  flex-wrap: wrap;
  align-content: space-between;
  position: relative;
  top: 0;
  transition: 300ms ease all;
  border: 1px solid ${C.GREY_LIGHTER};
  border-radius: 5px;
  overflow: hidden;

  &:hover {
    top: -10px;
    box-shadow: 0 4px 40px rgba(0, 0, 0, 0.15);

    ${PlaylistCardLink} {
      background-color: ${C.PRIMARY_COLOR_LIGHT};
      color: ${C.LIGHT};
      border-color: ${C.PRIMARY_COLOR_LIGHT};
    }
  }
`;

export const PlaylistCardImage = styled.div`
  img {
    display: block;
  }
`;

export const PlaylistCardContent = styled.div`
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  overflow: hidden;
  width: 100%;
`;

export const PlaylistCardInfo = styled.div`
  padding: ${SPACINGS.lg} ${SPACINGS.lg} 0;
`;

export const PlaylistCardAuthor = styled.p`
  color: ${C.GREY};
  ${T.BODY_TEXT}
`;

export const PlaylistCardAuthorDetail = styled.span`
  font-weight: bold;
`;

export const PlaylistCardTitle = styled.h2`
  color: ${C.GREY_DARK};
  margin: 0 auto ${SPACINGS.xs};
  ${T.TITLE_MEDIUM};
`;
