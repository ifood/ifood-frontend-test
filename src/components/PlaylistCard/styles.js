import styled from 'styled-components'

export const StyledPlaylistCard = styled.div`
  img {
    width: 100%;
    margin-bottom: 8px;
    box-shadow: 0px 24px 16px -16px rgba(0, 0, 0, 0.4);
    transition: ease 0.2s;
  }

  &:hover img {
    transform: scale(1.02);
  }
`
