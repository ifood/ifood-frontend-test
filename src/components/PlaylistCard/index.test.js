import React from "react";
import { render } from "@testing-library/react";

import PlaylistCard from "./index";

test("<PlaylistCard/> renders correcly", () => {
  const props = {
    link: "any_link",
    image: "any_url",
    title: "Today's Hitss",
    author: "Spotify",
    label: "Listen on spotify",
  };

  const component = <PlaylistCard {...props} />;
  const { asFragment } = render(component);
  expect(asFragment(component)).toMatchSnapshot();
});
