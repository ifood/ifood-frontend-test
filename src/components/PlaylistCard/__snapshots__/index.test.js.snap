// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaylistCard/> renders correcly 1`] = `
<DocumentFragment>
  <a
    class="sc-gtssRu hTbsDS"
    href="any_link"
    target="_blank"
  >
    <div
      class="sc-dlnjPT cDEBPx"
    >
      <img
        alt="Today's Hitss"
        src="any_url"
      />
    </div>
    <div
      class="sc-hKFyIo hQOELo"
    >
      <div
        class="sc-eCApGN bvuXSf"
      >
        <h2
          class="sc-iCoHVE hDdhGb"
        >
          Today's Hitss
        </h2>
        <p
          class="sc-jSFkmK fxJhem"
        >
          by 
          <span
            class="sc-gKAblj hnBNSm"
          >
            Spotify
          </span>
        </p>
      </div>
      <span
        class="sc-bdnylx bxgaUg"
      >
        Listen on spotify
      </span>
    </div>
  </a>
</DocumentFragment>
`;
